File: src/types/template.ts
export interface NoteTemplate {
  id: string;
  name: string;
  description: string;
  content: string;
  is_builtin: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateTemplateRequest {
  name: string;
  description: string;
  content: string;
}

export interface CreateNoteFromTemplateRequest {
  template_id: string;
  title: string;
}

File: src-tauri/src/types/template.rs
use serde::{Deserialize, Serialize};

/// Represents a note template
#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct NoteTemplate {
    pub id: String,
    pub name: String,
    pub description: String,
    pub content: String,
    pub is_builtin: bool,
    pub created_at: String,
    pub updated_at: String,
}

/// Request to create a template
#[derive(Debug, Deserialize, Serialize)]
pub struct CreateTemplateRequest {
    pub name: String,
    pub description: String,
    pub content: String,
}

/// Request to create a note from a template
#[derive(Debug, Deserialize, Serialize)]
pub struct CreateNoteFromTemplateRequest {
    pub template_id: String,
    pub title: String,
}
